sdks/java/core/src/main/java/org/apache/beam/sdk/util/UnownedOutputStream.java

```diff
@@ -53,4 +53,9 @@ public int hashCode() {
   public String toString() {
     return MoreObjects.toStringHelper(UnownedOutputStream.class).add("out", out).toString();
   }
+
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
+    out.write(b, off, len);
```

Comment:
Wondering if we should do the bounds check here too (like in `FilterOutputStream` or we assume that the delegated does the check? opinions @lukecwik

Reply:
Leave the bounds checking to the part of the system that accesses the `byte[]` otherwise every layer will perform the same validation.

Reply:
I agree with @lukecwik